### PR TITLE
Add a post about the architecture of WPE to the blog

### PR DESCRIPTION
**Pull request by csaavedra**
Also add a short bio page for myself

----

Site preview: https://igalia.github.io/wpewebkit.org/post-architecture-of-wpe/

### CODE REVIEW

**Comment by csaavedra** [comment]
@nikolaszimmermann the preview doesn't seem to work, it's my first time doing this so I literally have no idea what I'm doing :)

**Comment by bkardell** [comment]
> @nikolaszimmermann the preview doesn't seem to work, it's my first time doing this so I literally have no idea what I'm doing :)

The site preview link seems to work - it doesn't take you to the page automatically since you could, in fact, be adding or editing many pages and how would it know (in fact, you are because it gets added to the blog index too), so you have to navigate and check the pages.  Here is the preview of the post itself https://igalia.github.io/wpewebkit.org/post-architecture-of-wpe/blog/02-overview-of-wpe.html

File: _posts/2022-06-30-blog-overview-of-wpe.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+---
+layout: post
+title: "An overview of the WPE WebKit project"
+tags: [blogpost]
+author: csaavedra
+permalink: /blog/02-overview-of-wpe.html
+---
+
+In <a href="https://wpewebkit.org/blog/01-happy-birthday-wpe.html">the
```
Comment (aperezdc) [inline _posts/2022-06-30-blog-overview-of-wpe.md:9]: Could we make this link only have an absolute path? I just realized that's a good practice that plays better with the site preview feature we have for pull requests 🔨 

```suggestion
In <a href="/blog/01-happy-birthday-wpe.html">the
```

File: _posts/2022-06-30-blog-overview-of-wpe.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+API, making it easy for developers already familiar with the latter to
+start working with WPE. The cog library also serves as a wrapper
+around WPE to make it easier still. Once WPE was mature enough, it was
+accepted by Apple as an official WebKit port, meaning that the port
+lives in-tree and takes immediate advantage of the many improvements
+that land on the WebKit repository on a daily basis.
+
+## How does WPE integrate with WebKit?
+
+![A diagram of the WPE WebKit architecture](https://wpewebkit.org/assets/wpe-architecture-diagram.png)
```
Comment (aperezdc) [inline _posts/2022-06-30-blog-overview-of-wpe.md:93]: Same here:

```suggestion
![A diagram of the WPE WebKit architecture](/assets/wpe-architecture-diagram.png)
```

File: _posts/2022-06-30-blog-overview-of-wpe.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+- The <a href="https://github.com/WebPlatformForEmbedded/libwpe">libwpe</a> library. This is an out-of-tree library that provides the API required by the WPE port in a generic way to implement the graphical and input backends. Specific functionality for a concrete platform is not provided, but the library relies on the existence of a backend implementation, as is described next.
+- Finally, a WPE backend implementation. This is where all the platform-specific code lives. Backends are loadable modules that can be chosen depending on the underlying hardware. These should provide access to graphics and input depending on the specific architecture, platform, and operating system requirements. As a reference, <a href="https://github.com/Igalia/WPEBackend-fdo">WPEBackend-fdo</a> is a freedesktop.org-based backend, which uses Wayland and freekdesktop.org technologies, and is <a href="https://wpewebkit.org/about/supported-hardware.html">supported for several architectures</a>, including NXP and Broadcom chipsets, like the Raspberry PI, and also regular PC architectures, easing testing and development.
```
Comment (aperezdc) [inline _posts/2022-06-30-blog-overview-of-wpe.md:104]: And here:

```suggestion
- Finally, a WPE backend implementation. This is where all the platform-specific code lives. Backends are loadable modules that can be chosen depending on the underlying hardware. These should provide access to graphics and input depending on the specific architecture, platform, and operating system requirements. As a reference, <a href="https://github.com/Igalia/WPEBackend-fdo">WPEBackend-fdo</a> is a freedesktop.org-based backend, which uses Wayland and freekdesktop.org technologies, and is <a href="/about/supported-hardware.html">supported for several architectures</a>, including NXP and Broadcom chipsets, like the Raspberry PI, and also regular PC architectures, easing testing and development.
```

File: _posts/2022-06-30-blog-overview-of-wpe.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+An implementor interested in building a browser in a new architecture
+only needs to focus on the development of the last component -- a WPE
+backend. Having a backend, starting the development of a
+WebKit-powered browser is already much easier than it ever was!
+
+For a more detailed description of the architecture of WPE and WebKit,
+check this article on <a
+href="https://wpewebkit.org/about/architecture.html">the architecture
```
Comment (aperezdc) [inline _posts/2022-06-30-blog-overview-of-wpe.md:113]: One more:

```suggestion
href="/about/architecture.html">the architecture
```

Comment (aperezdc) [reply _posts/2022-06-30-blog-overview-of-wpe.md:113]: There's a couple more, I won't point them all one by one 😄

**Comment by csaavedra** [comment]
> Content LGTM, I only made a few comments about using links without hostnames (only the resource paths) so things play a bit nicer with the site preview feature.

Fixed those, thanks!

**Review by aperezdc — APPROVED** [review]
Thanks @csaavedra! I will leave the merging to you, in case you want to ask for more proofreading. From my point of view, you can merge whenever you feel it's a good moment 👍🏼

**Pull request merged**